packages/zoe/src/zoeService/zoe.js

```diff
@@ -0,0 +1,357 @@
+          return E(
+            /** @type Promise<addSeatObj> */ (addSeatObjPromiseKit.promise),
+          ).addSeat(invitationHandle, zoeSeatAdmin, seatData);
+        },
+        hasZoeSeatAdmin: zoeSeatAdmin => zoeSeatAdmins.has(zoeSeatAdmin),
+        removeZoeSeatAdmin: zoeSeatAdmin => zoeSeatAdmins.delete(zoeSeatAdmin),
+        getPublicFacet: () => publicFacetPromiseKit.promise,
+        getTerms: () => instanceRecord.terms,
+        getIssuers: () => instanceRecord.issuerKeywordRecord,
+        getBrands: () => instanceRecord.brandKeywordRecord,
+        getInstance: () => instance,
+      };
+
+      instanceToInstanceAdmin.init(instance, instanceAdmin);
+
+      /** @type {ZoeInstanceAdmin} */
+      const zoeInstanceAdminForZcf = {
+        makeInvitation: (invitationHandle, description, customProperties) => {
+          const invitationAmount = invitationAmountMath.make(
+            harden([
+              {
+                ...customProperties,
+                description,
+                handle: invitationHandle,
+                instance,
+                installation,
+              },
+            ]),
+          );
+          return invitationMint.mintPayment(invitationAmount);
+        },
+        // checks of keyword done on zcf side
+        saveIssuer: (issuerP, keyword) =>
+          issuerTable
+            .getPromiseForIssuerRecord(issuerP)
+            .then(({ issuer, brand }) => {
+              registerIssuerByKeyword(keyword, issuer, brand);
+              if (!brandToPurse.has(brand)) {
+                brandToPurse.init(brand, E(issuer).makeEmptyPurse());
+              }
+            }),
+        // A Seat requested by the contract without an offer
+        makeOfferlessSeat: (initialAllocation, proposal) => {
+          const { userSeat, notifier, zoeSeatAdmin } = makeZoeSeatAdminKit(
+            initialAllocation,
+            instanceAdmin,
+            cleanProposal(getAmountMath, proposal),
+            brandToPurse,
+          );
+          zoeSeatAdmins.add(zoeSeatAdmin);
+          return { userSeat, notifier, zoeSeatAdmin };
+        },
+        shutdown: () => {
+          exitAllSeats();
+          adminNode.terminate();
+        },
+        makeZoeMint,
+      };
+
+      // At this point, the contract will start executing. All must be ready
+
+      const {
+        creatorFacet = {},
+        publicFacet = {},
+        creatorInvitation,
+        addSeatObj,
+      } = await E(zcfRoot).executeContract(
+        bundle,
+        zoeService,
+        invitationIssuer,
+        zoeInstanceAdminForZcf,
+        harden({ ...instanceRecord }),
+      );
+
+      addSeatObjPromiseKit.resolve(addSeatObj);
+      publicFacetPromiseKit.resolve(publicFacet);
+
+      // creatorInvitation can be undefined, but if it is defined,
+      // let's make sure it is an invitation.
+      if (creatorInvitation !== undefined) {
+        assert(
+          await invitationIssuer.isLive(creatorInvitation),
```

Comment:
An `await` within an `assert` call argument, within an `if`. This is a double violation of our `await` restriction.
```suggestion
     const wellFormed = 
       await (creatorInvitation === undefined || invitationIssuer.isLive(creatorInvitation);
     // AWAIT
     assert(wellFormed,
```